legion_core/src/world.rs

```diff
@@ -731,6 +731,33 @@ impl World {
     /// Determines if the given `Entity` is alive within this `World`.
     pub fn is_alive(&self, entity: Entity) -> bool { self.entity_allocator.is_alive(entity) }
 
+    /// Returns the entity's component types, if the entity exists.
+    pub fn entity_component_types(
+        &self,
+        entity: Entity,
+    ) -> Option<&[(ComponentTypeId, ComponentMeta)]> {
+        if !self.is_alive(entity) {
+            return None;
+        }
+        let location = self.entity_locations.get(entity);
+        let archetype = location
+            .map(|location| self.storage().archetype(location.archetype()))
+            .unwrap_or(None);
+        archetype.map(|archetype| archetype.description().components())
+    }
+
+    /// Returns the entity's tag types, if the entity exists.
```

Comment:
Returns the tag id and meta is this equal to tag type? We could just name it  tag information

Reply:
If this tuple is used in more places we could also create a `type TagType = (TagTypeId, TagMeta);`